Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -202,6 +202,19 @@ public extension Sequence {
     func sorted<T: Comparable>(by keyPath: KeyPath<Element, T>) -> [Element] {
         return sorted { $0[keyPath: keyPath] < $1[keyPath: keyPath] }
     }
+
+    /// SwifterSwift: Sum of a `Numeric` property of each `Element` in a `Sequence`.
+    ///
+    ///     ["James", "Wade", "Bryant"].sum(for: \.count) -> 15
+    ///
+    /// - Parameter keyPath: Key path of the `Numeric` property.
+    /// - Returns: The sum of the `Numeric` propertys at `keyPath`.
+    func sum<T: Numeric>(for keyPath: KeyPath<Element, T>) -> T {
+        // Implementation from: https://swiftbysundell.com/articles/reducers-in-swift/
+        return reduce(0) { sum, element in
```

Comment:
I'm generally a proponent of building on top of existing solutions.  SwifterSwift already implements `sum()` so I would utilize that:

`return map { $0[keyPath: keyPath] }.sum()`

Reply:
SwifterSwift is designed to be copyable piecemeal, so we can't reference our own functions between themselves. Otherwise, anybody wanting to copy a function manually may find themselves traversing one function after the next. It's a big drawback, but it makes the library more flexible.

Reply:
I found the following rule in the contributing guidelines:
> - Do not use an existing SwifterSwift extension inside another SwifterSwift extension. All extensions should be able to be copied and pasted and work immediately without having to copy another extension.

Let me know if I should change the implementation to your proposed solution regardless :)

Reply:
If the intention is each _function_ stand on its own then I agree.  Perhaps the guidelines could make that a little clearer (function vs extension).

That said, I might argue anyone copying implementations out of your library isn't _using your library_ so it's not really "more flexible."  ;)

Reply:
Yeah, I realised straight away that "flexible" was the wrong word, but you get me ;)